### ISSUE TITLE
4.0.0 build 42 changes xarcade button mapping

### ISSUE THREAD

**Issue by BestUsername** [recalbox/recalbox-os#818]
### What happened
xarcade joystick button mapping changed

### What should have happened
I have xarcade controllers enabled in config, and all others disabled. The xarcade service still picks up the device and maps it to joypads, but all of the button mappings have changed and it's unusable for gaming in this state. This worked in previous 4.0.0 builds

### Enumerate your devices
    8.686905] usb 1-1.5: New USB device found, idVendor=aa55, idProduct=0101
[    8.686934] usb 1-1.5: New USB device strings: Mfr=1, Product=2, SerialNumber=0
[    8.686946] usb 1-1.5: Product: X-Arcade
[    8.686955] usb 1-1.5: Manufacturer: XGaming
[    8.727815] input: XGaming X-Arcade as /devices/platform/soc/3f980000.usb/usb1/1-1/1-1.5/1-1.5:1.0/0003:AA55:0101.0001/input/input0
[    8.784047] hid-generic 0003:AA55:0101.0001: input,hidraw0: USB HID v1.10 Keyboard [XGaming X-Arcade] on usb-3f980000.usb-1.5/input0
[    8.818756] input: XGaming X-Arcade as /devices/platform/soc/3f980000.usb/usb1/1-1/1-1.5/1-1.5:1.1/0003:AA55:0101.0002/input/input1
[    8.874492] hid-generic 0003:AA55:0101.0002: input,hidraw1: USB HID v1.10 Keyboard [XGaming X-Arcade] on usb-3f980000.usb-1.5/input1

### How to reproduce the bug
update to RecalBox 4.0.0 build 42

**Comment by substring** [comment]
Hey Marc

Thank you for your feedback.A few buttons chnaged name to have something that look like a 21st century pad.
Also, some buttons changed swapped their functions for a better.compatibility with the default MAME input settings and keyboard encoders.

2 solutions :
- either remap your buttons in ES
- either `cp /recalbox/share_init/system/.emulationstation/es_input.cfg ~/.emulationstation`

Might revert back to the previous buttons name for a few of them, you're on a development branch ;)

**Comment by BestUsername** [comment]
Thank you for the quick reply!
I followed the second step, and the player 2 controls seem to be back to normal, but the system mappings (retroarch menu, special button, coin input) are still wonky and overlap player 1 controls.
Is the easiest course of action to re-install 4.0.0 beta and set upgrades to stable?

**Comment by substring** [comment]
Could you explain what you mean by "overlapping" compared to before ? Player 2 is not supposed to have a "valid" hotkey as it's never used.

**Comment by BestUsername** [comment]
Apologies - I wasn't clear.
Player 2 controls were fine.
The special commands (I'm not sure what to call them both) overlapped player 1 controls, and the coin button was still different.
If the xarcade buttons for player one per row are:
1-4, 5-8, 9-10
And the two buttons on the left side are
11(top) and 12(bottom)
Then button 5 and 9 both brought up the retroarch menu. The coin button also changed from 12 to 10, but the retroarch menu is the big one. Not only does it prevent a player from using that button in games, but when in the retroarch menu button 5 is the "back" button. But instead of going back, it closes the menu. So you can only drill down in the retroarch menu.

I have since re-installed 4.0.0 beta2 and everything is great again. But if it would help, I can keep a separate microsd to help test the xarcade hardware with dev builds.

**Comment by substring** [comment]
First things first :
- Hotkey is now the flipper butto
- SELECT is the button just right to the previous hotkey (KEY_5, see http://www.xgaming.com/service/images/Layouts/PC-MAC_Mode_1_Button_Layout.jpg)
- some buttons changed their name
- had to map default IPAC2 keys because beyond a a limit, MAME keys are not standard naymore

If you were to install build42 on a blank system, you wouldn't have any problem. The difficulty is that I don't own a xarcade, so I have to use my keyboard and send xarcade keystrokes ...

Anyway, the mapping having changed quite much makes updates a mess, so i may revert back some changes, not all of them

**Comment by BestUsername** [comment]
Right on. 
I wonder if there are any problems with using the xarcade service that remaps the tankstick to 2 joypads (or are you testing with your keyboard through that service?)
Either way, I'll mark this as closed because it currently only deals with upgrading through unstable/testing.
And as I offered before, let me know if you want me to test dev builds with the tankstick hardware.

**Comment by substring** [comment]
well yeah, i need feedback on how it's mapped for beta3, which means testing the unstable branch you're on now :)

**Comment by BestUsername** [comment]
Sorry for the delay - I was unable to test the upgrade path properly because it kept switching between upgrading and downgrading back and forth.

This morning I got an upgrade notice on my stable version, and I ran into the same issues.
Remembering you said it was an upgrade path problem, I then installed 4.0.0 beta3 fresh on to a micrsd.
I have the same problems as before, so this isn't an upgrade issue.

Are there any tests you'd like me to run on my fresh 4.0.0 beta3 install?
